### PR TITLE
Update oldschoolkeys, tabmenu, yanksel plugins

### PR DESCRIPTION
**Pull request by schvabodka-man**
Hello there! Due to changes in API plugin tabmenu was broken and unusable on newer versions of luakit. I fixed it to work with new api and it works nice. I've also removed standart hotkey for it(because anyway you can set it in you luakit config and not everyone wants to have this hotkey). I also fixed yankselect and old school keybindings plugins. Would be sweat if you'll merge this

### CODE REVIEW

File: oldschoolkeys/init.lua [hunk 1]
```diff
@@ -4,24 +4,16 @@
...
-    -- Normal mode:
-    normal = {
-        key({},          "b",           function (w, m) w:back(m.count)    end, {count=1}),
-        key({"Mod1"},    "Page_Up",     function (w, m) w.tabs:reorder(w.view, w.tabs:current() - m.count) end, {count=1}),
-        key({"Mod1"},    "Page_Down",   function (w, m) w.tabs:reorder(w.view, (w.tabs:current() + m.count) % w.tabs:count()) end, {count=1}),
-    },
-}
+local add_binds = modes.add_binds
+
+module("plugins.oldschoolkeys")
```
Comment (aidanholm) [inline oldschoolkeys/init.lua]: The use of `module()` in plugins is now forbidden. This is because it unavoidably sets a global variable. Instead, declare a local table named `_M` and return it at the end of the file. All subsequent top-level variables should either be `local` or should be fields of `_M`.

File: oldschoolkeys/init.lua [hunk 2]
```diff
@@ -4,24 +4,16 @@
 
 local pairs = pairs
```
Comment (aidanholm) [inline oldschoolkeys/init.lua]: Remove this

File: oldschoolkeys/init.lua [hunk 3]
```diff
@@ -4,24 +4,16 @@
...
-        key({},          "b",           function (w, m) w:back(m.count)    end, {count=1}),
-        key({"Mod1"},    "Page_Up",     function (w, m) w.tabs:reorder(w.view, w.tabs:current() - m.count) end, {count=1}),
-        key({"Mod1"},    "Page_Down",   function (w, m) w.tabs:reorder(w.view, (w.tabs:current() + m.count) % w.tabs:count()) end, {count=1}),
-    },
-}
+local add_binds = modes.add_binds
+
+module("plugins.oldschoolkeys")
+
+add_binds("normal",
```
Comment (aidanholm) [inline oldschoolkeys/init.lua]: Opening brace should go on the same line as `add_binds`, and closing brace+paren should go on a new line:
```lua
add_binds("normal", {
   ...
})
```

File: tabmenu/init.lua [hunk 4]
```diff
@@ -7,80 +7,71 @@ local ipairs = ipairs
 local table = table
 
 local lousy = require "lousy"
+
```
Comment (aidanholm) [inline tabmenu/init.lua]: Remove space here

File: tabmenu/init.lua [hunk 5]
```diff
@@ -7,80 +7,71 @@ local ipairs = ipairs
...
+local modes = require "modes"
+
```
Comment (aidanholm) [inline tabmenu/init.lua]: ditto

File: tabmenu/init.lua [hunk 6]
```diff
@@ -7,80 +7,71 @@ local ipairs = ipairs
...
-            table.insert(rows, {escape(view.uri), escape(view.title), v = view })
-        end
-        w.menu:build(rows)
-        local cur = w.tabs:current()
-        local ind = 0
-        repeat w.menu:move_down(); ind = ind + 1 until ind == cur
-        w.sbar.ebox:show()
-        w:notify("Use j/k to move, d close, Return switch.", false)
-    end,
+			enter = function (w)
```
Comment (aidanholm) [inline tabmenu/init.lua]: use spaces for indentation, not tabs

File: tabmenu/init.lua [hunk 7]
```diff
@@ -7,80 +7,71 @@ local ipairs = ipairs
...
-}, menu_binds))
-
--- Add key binds.
-local buf = lousy.bind.buf
-add_binds("normal", {
-    key({},     "m",    function (w)
-        w:set_mode("tabmenu")
-    end),
+add_binds("tabmenu", {
+			 { "<Delete>", "Delete tab.", function (w)
```
Comment (aidanholm) [inline tabmenu/init.lua]: This should only be indented by four spaces

File: yanksel/init.lua [hunk 8]
```diff
@@ -5,18 +5,22 @@
 local error = error
```
Comment (aidanholm) [inline yanksel/init.lua]: Remove these two lines

File: yanksel/init.lua [hunk 9]
```diff
@@ -5,18 +5,22 @@
...
 local luakit = luakit
 local lousy = require("lousy")
-local key, buf, but = lousy.bind.key, lousy.bind.buf
-local add_binds, add_cmds = add_binds, add_cmds
+
```
Comment (aidanholm) [inline yanksel/init.lua]: remove space

File: yanksel/init.lua [hunk 10]
```diff
@@ -5,18 +5,22 @@
...
+local modes = require "modes"
+
```
Comment (aidanholm) [inline yanksel/init.lua]: ditto

File: yanksel/init.lua [hunk 11]
```diff
@@ -5,18 +5,22 @@
...
+local add_binds = modes.add_binds
+local add_cmds = modes.add_cmds
 
 module("plugins.yanksel")
```
Comment (aidanholm) [inline yanksel/init.lua]: ditto re module

File: yanksel/init.lua [hunk 12]
```diff
@@ -5,18 +5,22 @@
...
-    buf("^ys$",
-	function (w)
-	    local text = luakit.selection.primary
-	    if not text then w:error("Empty selection.") return end
-	    luakit.selection.clipboard = text
-	    w:notify("Yanked selection: " .. text)
-	    luakit.selection.primary = ""
-	end),
-})
+function yank_select(w)
```
Comment (aidanholm) [inline yanksel/init.lua]: I'd say make this an action table, which is just a table with `func` (function) and `desc` (description) keys. See `lib/binds.lua` for examples (scrolling actions).

File: tabmenu/init.lua [hunk 13]
```diff
@@ -7,80 +7,71 @@ local ipairs = ipairs
...
+local modes = require "modes"
+
+local add_binds = modes.add_binds
+local add_cmds = modes.add_cmds
+local new_mode = modes.new_mode
+
 local add_binds, add_cmds = add_binds, add_cmds
 local new_mode, menu_binds = new_mode, menu_binds
 
 module("plugins.tabmenu")
```
Comment (aidanholm) [inline tabmenu/init.lua]: ditto re module

File: tabmenu/init.lua [hunk 14]
```diff
@@ -7,80 +7,71 @@ local ipairs = ipairs
...
 
 hide_box = false
```
Comment (aidanholm) [inline tabmenu/init.lua]: make this `_M.hide_box`

**Comment by schvabodka-man** [comment]
Okay, here it goes. Check this one. Maybe in few time i'll fix user agent changer

File: oldschoolkeys/init.lua [hunk 15]
```diff
@@ -1,19 +1,17 @@
 ------------------------------------------------------------------------
 -- Add some convenient keybindings back.                              --
 ------------------------------------------------------------------------
-
-local pairs = pairs
 local lousy = require("lousy")
```
Comment (aidanholm) [inline oldschoolkeys/init.lua]: No blank lines between require lines

File: oldschoolkeys/init.lua [hunk 16]
```diff
@@ -1,19 +1,17 @@
...
-add_binds("normal",
-		  {{ "b", "Back", function (w, m) w:back(m.count) end, {count=1}},
-			 { "<Mod1-Page_Up>", "Reorder tabs", function (w, m) w.tabs:reorder(w.view, w.tabs:current() - m.count) end, {count=1}},
-			 { "<Mod1-Page_Down>", "Reorder tabs", function (w, m) w.tabs:reorder(w.view, (w.tabs:current() + m.count) % w.tabs:count()) end, {count=1}}})
+local _M = {}
 
+add_binds("normal",{
+    { "b", "Back", function (w, m) w:back(m.count) end, {count=1}},
+    { "<Mod1-Page_Up>", "Reorder tabs", function (w, m) w.tabs:reorder(w.view, w.tabs:current() - m.count) end, {count=1}},
+    { "<Mod1-Page_Down>", "Reorder tabs", function (w, m) w.tabs:reorder(w.view, (w.tabs:current() + m.count) % w.tabs:count()) end, {count=1}}})
```
Comment (aidanholm) [inline oldschoolkeys/init.lua]: Final `})` should go on a new line

File: oldschoolkeys/init.lua [hunk 17]
```diff
@@ -1,19 +1,17 @@
...
-
-add_binds("normal",
-		  {{ "b", "Back", function (w, m) w:back(m.count) end, {count=1}},
-			 { "<Mod1-Page_Up>", "Reorder tabs", function (w, m) w.tabs:reorder(w.view, w.tabs:current() - m.count) end, {count=1}},
-			 { "<Mod1-Page_Down>", "Reorder tabs", function (w, m) w.tabs:reorder(w.view, (w.tabs:current() + m.count) % w.tabs:count()) end, {count=1}}})
+local _M = {}
 
+add_binds("normal",{
+    { "b", "Back", function (w, m) w:back(m.count) end, {count=1}},
+    { "<Mod1-Page_Up>", "Reorder tabs", function (w, m) w.tabs:reorder(w.view, w.tabs:current() - m.count) end, {count=1}},
```
Comment (aidanholm) [inline oldschoolkeys/init.lua]: These functions are long enough that they should be multi-line

File: yanksel/init.lua [hunk 18]
```diff
@@ -1,26 +1,29 @@
...
-
 local modes = require "modes"
-
 local add_binds = modes.add_binds
 local add_cmds = modes.add_cmds
 
-module("plugins.yanksel")
+local _M = {}
+
+local funcs = {
```
Comment (aidanholm) [inline yanksel/init.lua]: This table should be called `actions`

**Comment by schvabodka-man** [comment]
It's passing tests

**Comment by aidanholm** [comment]
I'm getting the following test failures:
```
FAIL style/test_luacheck / test_luacheck
  tests/style/test_luacheck.lua:102: Luacheck messages:
    config/tabmenu/init.lua:28: (9:16):                setting non-standard global variable 'hide_box'
    config/tabmenu/init.lua:42: (12:19):               accessing undefined variable 'hide_box'
    config/oldschoolkeys/init.lua:4: (7:11):           unused variable 'lousy'
    config/yanksel/init.lua:5: (7:11):                 unused variable 'lousy'
    config/yanksel/init.lua:12: (18:18):               line contains trailing whitespace
FAIL style/test_source_format / test_vim_modeline
  tests/style/test_source_format.lua:41: Some files do not have modelines:
  config/tabmenu/init.lua                                Missing/malformed modeline
  config/oldschoolkeys/init.lua                          Missing/malformed modeline
  config/yanksel/init.lua                                Missing/malformed modeline
FAIL style/test_source_format / test_no_trailing_whitespace
  tests/style/test_source_format.lua:313: Some files have trailing whitespace:
  config/yanksel/init.lua:12                             Trailing whitespace
FAIL style/test_source_format / test_no_tabs_in_indentation
  tests/style/test_source_format.lua:287: Some files have tabs in indentation:
  config/tabmenu/init.lua:20                             Tabs in indentation
  config/oldschoolkeys/init.lua:14                       Tabs in indentation
  config/oldschoolkeys/init.lua:15                       Tabs in indentation
  config/oldschoolkeys/init.lua:16                       Tabs in indentation
  config/oldschoolkeys/init.lua:18                       Tabs in indentation
  config/oldschoolkeys/init.lua:19                       Tabs in indentation
  config/oldschoolkeys/init.lua:20                       Tabs in indentation
  config/yanksel/init.lua:14                             Tabs in indentation
  config/yanksel/init.lua:15                             Tabs in indentation
  config/yanksel/init.lua:16                             Tabs in indentation
  config/yanksel/init.lua:17                             Tabs in indentation
  config/yanksel/init.lua:18                             Tabs in indentation
  config/yanksel/init.lua:19                             Tabs in indentation
  config/yanksel/init.lua:20                             Tabs in indentation
  config/yanksel/init.lua:21                             Tabs in indentation
```

The trailing whitespace / tabs are easy to fix. Just remove the lines with unused variables.
For the modelines, you need to add the following to the bottom of each file, with a blank line above it:
```lua
-- vim: et:sw=4:ts=8:sts=4:tw=80
```

**Comment by aidanholm** [comment]
On tabmenu.lua lines 28 and 42, `hide_box` should be `_M.hide_box`; that should be everything!

**Comment by schvabodka-man** [comment]
I'm not using vim, i'm using emacs. I just replaced tabs with spaces(4 tabs = 4 spaces) and run it through lua beautifier. Hide box is already _M.hide_box(really it is). And 28 and 42 is intended with 4 spaces

**Comment by schvabodka-man** [comment]
Ok, wait for a second i fixed those references

**Comment by aidanholm** [comment]
Even so, the project's coding standard is to use vim modelines (and I thought emacs could read vim modelines?)
I'm guessing perhaps you forgot to push changes then? I have the same version as on this PR.

**Comment by aidanholm** [comment]
Ok, I see your fixes, let me try again

**Comment by schvabodka-man** [comment]
ok, i pushed, some more fixes. should be ok with this variables. And i know shit about these "cool" vim aliases. My emacs display these lines pretty nicely. Maybe in vim the're broken somehow
https://transfer.sh/14MtiM/screenshot.png

**Comment by aidanholm** [comment]
Ok, references are fixed, it's just whitespace errors and the extra `require "lousy"` lines now; once these are fixed it should be ready to merge!

**Comment by aidanholm** [comment]
Nice desktop btw; which wm?

**Comment by schvabodka-man** [comment]
I think it would be ok to merge all this fixes commits into three comits or not?

I3wm, polybar with bunch of scripts, rofi, Fedora(but most likely i would migrate to Void), fish shell and Emacs

**Comment by Plaque-fcc** [comment]
If they can be divided into three logically solid changesets — yes.

On August 21, 2017 7:35:29 PM GMT+03:00, Ivan <notifications@github.com> wrote:
>I think it would be ok to merge all this fixes commits into three
>comits or not?
>
>I3wm, polybar with bunch of scripts, rofi, Fedora(but most likely i
>would migrate to Void), fish shell and Emacs
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/luakit/luakit-plugins/pull/14#issuecomment-323793382

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by schvabodka-man** [comment]
I don't understand about which extra require "lousy" you're talking about. Projectile with grep gives me 4 require lousy(each for 1 plugin)

**Comment by aidanholm** [comment]
- I'd also prefer that there be three commits, "Update tabmenu module" , "Update yanksel module", etc.
- oldschoolkeys and yanksel never use the lousy variable, so it should be removed.

**Comment by schvabodka-man** [comment]
Fixed that one lousy

**Comment by aidanholm** [comment]
Just whitespace issues now; by the way once this is finished it'd be good to modify the README as well to say which plugins are updated and which aren't.

**Comment by schvabodka-man** [comment]
Can you point me on those whitespaces?

**Comment by aidanholm** [comment]
Have you ran the tests? That'll show you where the errors are, but they should be basically the same as in the test paste above. I'd just run `sed -e 's/\t/    /g' -e 's/ $//'` on `oldschoolkeys/init.lua` and `yanksel/init.lua`

Separately, the files are still missing modelines

**Comment by schvabodka-man** [comment]
I've run this sed command but it hasn't changed anything(diff is null).

**Comment by schvabodka-man** [comment]
Okay i've added modelines and i still don't know what whitespaces you're talking about

**Comment by aidanholm** [comment]
Sorry, I wasn't very clear when explaining before. There are two whitespace issues:

- The lua files have tabs at the beginning of the line, as part of the indentation. This goes against the luakit code style standard, which is that all files must be indented with spaces only.

  To find these files, run `grep -rPn '\t' tabmenu oldschoolkeys yanksel` from the `luakit-plugins` root folder. The easiest way to fix these problems is to do a find and replace in emacs, but you can also use the sed command I posted before.

  Here is the output I get, that shows where the problems are:
```
tabmenu/init.lua:20:	  { ":tabmenu", [[Open tab menu]], function (w) w:set_mode("tabmenu") end },
oldschoolkeys/init.lua:13:	  function (w, m)
oldschoolkeys/init.lua:14:		 w.tabs:reorder(w.view, w.tabs:current() - m.count)
oldschoolkeys/init.lua:15:	  end, {count=1}},
oldschoolkeys/init.lua:17:	  function (w, m)
oldschoolkeys/init.lua:18:		 w.tabs:reorder(w.view, (w.tabs:current() + m.count) % w.tabs:count())
oldschoolkeys/init.lua:19:	  end, {count=1}}
yanksel/init.lua:13:	  desc = "Yank selection.",
yanksel/init.lua:14:	  func = function (w)
yanksel/init.lua:15:		 local text = luakit.selection.primary
yanksel/init.lua:16:		 if not text then w:error("Empty selection.") return end
yanksel/init.lua:17:		 luakit.selection.clipboard = text
yanksel/init.lua:18:		 w:notify("Yanked selection: " .. text)
yanksel/init.lua:19:		 luakit.selection.primary = ""
yanksel/init.lua:20:	  end,
```

- Trailing whitespace; there's a space at the end of a line in the `yanksel/init.lua` file. This is also not allowed by the current code standards, as it makes diffs larger than necessary, which makes it more difficult to read them and can cause issues with `git blame`.

  The problem is at line 11 of `yanksel/init.lua` (`local actions = {`).

Let me know if that's not clear!

**Comment by Plaque-fcc** [comment]
Guess it was my bad on tab indents. :<

On August 23, 2017 2:56:40 PM GMT+03:00, Aidan Holm <notifications@github.com> wrote:
>Sorry, I wasn't very clear when explaining before. There are two
>whitespace issues:
>
>- The lua files have tabs at the beginning of the line, as part of the
>indentation. This goes against the luakit code style standard, which is
>that all files must be indented with spaces only.
>
>To find these files, run `grep -rPn '\t' tabmenu oldschoolkeys yanksel`
>from the `luakit-plugins` root folder. The easiest way to fix these
>problems is to do a find and replace in emacs, but you can also use the
>sed command I posted before.
>
>  Here is the output I get, that shows where the problems are:
>```
>tabmenu/init.lua:20:	  { ":tabmenu", [[Open tab menu]], function (w)
>w:set_mode("tabmenu") end },
>oldschoolkeys/init.lua:13:	  function (w, m)
>oldschoolkeys/init.lua:14:		 w.tabs:reorder(w.view, w.tabs:current() -
>m.count)
>oldschoolkeys/init.lua:15:	  end, {count=1}},
>oldschoolkeys/init.lua:17:	  function (w, m)
>oldschoolkeys/init.lua:18:		 w.tabs:reorder(w.view, (w.tabs:current() +
>m.count) % w.tabs:count())
>oldschoolkeys/init.lua:19:	  end, {count=1}}
>yanksel/init.lua:13:	  desc = "Yank selection.",
>yanksel/init.lua:14:	  func = function (w)
>yanksel/init.lua:15:		 local text = luakit.selection.primary
>yanksel/init.lua:16:		 if not text then w:error("Empty selection.")
>return end
>yanksel/init.lua:17:		 luakit.selection.clipboard = text
>yanksel/init.lua:18:		 w:notify("Yanked selection: " .. text)
>yanksel/init.lua:19:		 luakit.selection.primary = ""
>yanksel/init.lua:20:	  end,
>```
>
>- Trailing whitespace; there's a space at the end of a line in the
>`yanksel/init.lua` file. This is also not allowed by the current code
>standards, as it makes diffs larger than necessary, which makes it more
>difficult to read them and can cause issues with `git blame`.
>
> The problem is at line 11 of `yanksel/init.lua` (`local actions = {`).
>
>Let me know if that's not clear!
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/luakit/luakit-plugins/pull/14#issuecomment-324306418

-- 
J.K.

**Comment by aidanholm** [comment]
@Plaque-fcc no worries, there weren't any tests for that kind of thing with the old luakit, and most things had a mix of tabs and spaces :)

**Comment by Plaque-fcc** [comment]
No, it was my first meet of Lua-driven software. And I still thank @mason-larobina for this.

On August 23, 2017 3:00:48 PM GMT+03:00, Aidan Holm <notifications@github.com> wrote:
>@Plaque-fcc no worries, there weren't any tests for that kind of thing
>with the old luakit, and most things had a mix of tabs and spaces :) 
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/luakit/luakit-plugins/pull/14#issuecomment-324307321

-- 
J.K.

**Comment by aidanholm** [comment]
Hey @schvabodka-man, any progress on this one?

**Comment by schvabodka-man** [comment]
No, progress. I'm busy now

**Comment by aidanholm** [comment]
No problem

**Comment by schvabodka-man** [comment]
Maybe in 2-3 days i will actually start to work on these whitespaces

**Comment by aidanholm** [comment]
I've merged this one manually; thanks @schvabodka-man for updating these :)

**Comment by schvabodka-man** [comment]
Thx, i really appreciate your cleanup on whitespaces cause i don't have much time now

**Comment by aidanholm** [comment]
No problem
